src/backend/session/dbus/logind.rs

```diff
@@ -611,3 +555,91 @@ impl AsErrno for Error {
+                    std::io::ErrorKind::NotFound,
+                    "Neither libsystemd.so not libelogind.so were found",
+                )))
+            }
+        };
+
+        let session_name = unsafe {
+            let mut session_name: *mut c_char = std::ptr::null_mut();
+            let ret = (lib_logind.sd_pid_get_session)(0, &mut session_name);
+            if ret < 0 {
+                return Err(super::Error::FailedToGetSession(
+                    std::io::Error::from_raw_os_error(-ret),
+                ));
+            }
+            CString::from_raw(session_name)
+        };
+
+        let seat_name = unsafe {
+            let mut seat_name: *mut c_char = std::ptr::null_mut();
+            let ret = (lib_logind.sd_session_get_seat)(session_name.as_ptr(), &mut seat_name);
+            if ret < 0 {
+                return Err(super::Error::FailedToGetSeat(std::io::Error::from_raw_os_error(
+                    -ret,
+                )));
+            }
+            CString::from_raw(seat_name)
+        };
+
+        let vt_num = unsafe {
+            let mut vt_num = 0;
+            let ret = (lib_logind.sd_session_get_vt)(session_name.as_ptr(), &mut vt_num);
+            if ret < 0 {
+                None
+            } else {
+                Some(vt_num)
+            }
+        };
+
+        Ok((
+            session_name.into_string().unwrap(),
+            seat_name.into_string().unwrap(),
+            vt_num,
+        ))
+    }
+    struct LibLogind {
+        _lib: Library,
+        sd_pid_get_session: Symbol<unsafe extern "C" fn(pid_t, *mut *mut c_char) -> c_int>,
+        sd_session_get_seat: Symbol<unsafe extern "C" fn(*const c_char, *mut *mut c_char) -> c_int>,
+        sd_session_get_vt: Symbol<unsafe extern "C" fn(*const c_char, *mut c_uint) -> c_int>,
+    }
+
+    lazy_static::lazy_static! {
+        static ref LIB_LOGIND: Option<LibLogind> = {
+            // attempt to load libsystemd.so, if failure, attempt to load libelogind.so
+            Library::new("libsystemd.so").or_else(|_| Library::new("libelogind.so")).ok().and_then(|lib| {
+                unsafe {
+                    let sd_pid_get_session = match lib.get::<unsafe extern "C" fn(pid_t, *mut *mut c_char) -> c_int>(b"sd_pid_get_session\0") { Ok(s) => s.into_raw(), Err(_) => return None };
+                    let sd_session_get_seat = match lib.get::<unsafe extern "C" fn(*const c_char, *mut *mut c_char) -> c_int>(b"sd_session_get_seat\0") { Ok(s) =>  s.into_raw(), Err(_) => return None };
+                    let sd_session_get_vt = match lib.get::<unsafe extern "C" fn(*const c_char, *mut c_uint) -> c_int>(b"sd_session_get_vt\0") { Ok(s) => s.into_raw(), Err(_) => return None };
+                    Some(LibLogind {
+                        _lib: lib,
+                        sd_pid_get_session,
+                        sd_session_get_seat,
+                        sd_session_get_vt
```

Comment:
Is there any benefit of doing that instead of normal dynamic linking? IIRC those libraries could be ABI incompatible and there's no way they could be in the same time on the same system. I feel like just addign pkg config at build.rs could make sense.